buildkite/resource_pipeline_schedule.go

```diff
@@ -191,37 +232,57 @@ func (ps *pipelineSchedule) Read(ctx context.Context, req resource.ReadRequest,
 func (ps *pipelineSchedule) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
 	var state, plan pipelineScheduleResourceModel
 
-	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	diagsState := req.State.Get(ctx, &state)
+	diagsPlan := req.Plan.Get(ctx, &plan)
+
+	resp.Diagnostics.Append(diagsState...)
+	resp.Diagnostics.Append(diagsPlan...)
 
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	envVars := envVarsMapFromTfToString(ctx, plan.Env)
-	input := PipelineScheduleUpdateInput{
-		Id:       state.Id.ValueString(),
-		Label:    plan.Label.ValueStringPointer(),
-		Cronline: plan.Cronline.ValueStringPointer(),
-		Branch:   plan.Branch.ValueStringPointer(),
-		Commit:   plan.Commit.ValueStringPointer(),
-		Message:  plan.Message.ValueStringPointer(),
-		Env:      &envVars,
-		Enabled:  plan.Enabled.ValueBool(),
+	timeouts, diags := ps.client.timeouts.Update(ctx, DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
 	}
 
-	log.Printf("Updating Pipeline schedule %s ...", plan.Label.ValueString())
-	_, err := updatePipelineSchedule(ctx,
-		ps.client.genqlient,
-		input,
-	)
+	err := retry.RetryContext(ctx, timeouts, func() *retry.RetryError {
+		var err error
+
+		envVars := envVarsMapFromTfToString(ctx, plan.Env)
+		input := PipelineScheduleUpdateInput{
+			Id:       state.Id.ValueString(),
+			Label:    plan.Label.ValueStringPointer(),
+			Cronline: plan.Cronline.ValueStringPointer(),
+			Branch:   plan.Branch.ValueStringPointer(),
+			Commit:   plan.Commit.ValueStringPointer(),
+			Message:  plan.Message.ValueStringPointer(),
+			Env:      &envVars,
+			Enabled:  plan.Enabled.ValueBool(),
+		}
```

Comment:
Not a big deal, but this would be called over and over again and create a new struct each time it retries. Leaving it outside this function would be more performant. But we are talking nano seconds really so not worth worrying about - just letting you know

Reply:
You're right. I have moved this out of the retry loop